src/tools/immutable_object_cache/Types.cc

```diff
@@ -78,6 +79,7 @@ void ObjectCacheReadData::encode_payload() {
   ceph::encode(read_len, payload);
   ceph::encode(pool_id, payload);
   ceph::encode(snap_id, payload);
+  ceph::encode(object_size, payload);
```

Comment:
You cannot inject new fields in the middle and maintain backwards compatibility. Plus, you also need to increment the version.